ssl/s3_lib.c

```diff
@@ -3825,12 +3825,24 @@ long ssl3_ctrl(SSL *s, int cmd, long larg, void *parg)
     case SSL_CTRL_GET_CHAIN_CERT_STORE:
         return ssl_cert_get_cert_store(sc->cert, parg, 1);
 
+    case SSL_CTRL_GET_PEER_SIGNATURE_NAME:
+        if (sc->s3.tmp.peer_sigalg == NULL)
```

Comment:
I would NULL check before dereference - but I note that most of the setters here do and most of the getters do not.
So this is just a suggestion ...

Reply:
I am only dereferencing `sc` which is checked on entry into the function:
https://github.com/openssl/openssl/blob/364a7067de7c0f4e09fc220336f258c87011e836/ssl/s3_lib.c#L3537-L3538

Reply:
`*(const char **)parg `

I didn't tag the precise line - but it is the parg dereference